CoursesManager.UI/Models/Course.cs

```diff
@@ -1,9 +1,12 @@
-﻿using System.Collections.ObjectModel;
+﻿using CoursesManager.MVVM.Data;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
 using System.Text;
+using System.Windows.Media.Imaging;
 
 namespace CoursesManager.UI.Models
 {
-    public class Course
+    public class Course : IsObservable, ICopyable<Course>, IDataErrorInfo
```

Comment:
In principe zit in ViewModel al een INotifyDataErrorInfo implementatie. Hoef je niet per se te gebruiken maar val wel onder een van de evl's dus moet je wel een geldige reden voor hebben.